docs/_posts/2019-11-16-signal.md

```diff
@@ -0,0 +1,77 @@
+  Running: false
+  StartedAt: "2019-11-12T21:47:13.9406019Z"
+  Status: exited
+wash . ❯ signal docker/containers/wash_tutorial_redis_1 start
+wash . ❯ meta docker/containers/wash_tutorial_redis_1
+...
+State:
+  Dead: false
+  Error: ""
+  ExitCode: 0
+  FinishedAt: "2019-11-16T18:58:03.7284542Z"
+  OOMKilled: false
+  Paused: false
+  Pid: 63392
+  Restarting: false
+  Running: true
+  StartedAt: "2019-11-16T18:58:49.6323676Z"
+  Status: running
+```
+
+If the entry's schema is known, then you can use the `docs` command to view its supported signals/signal groups.
+
+```
+wash . ❯ docs docker/containers/wash_tutorial_redis_1
+No description provided.
+
+SUPPORTED SIGNALS
+* start
+    Starts the container. Equivalent to 'docker start <container>'
+* stop
+    Stops the container. Equivalent to 'docker stop <container>'
+* pause
+    Suspends all processes in the container. Equivalent to 'docker pause <container>'
+* resume
+    Un-suspends all processes in the container. Equivalent to 'docker unpause <container>'
+* restart
+    Restarts the container. Equivalent to 'docker restart <container>'
+
+SUPPORTED SIGNAL GROUPS
+* linux
+    Consists of all the supported Linux signals like SIGHUP, SIGKILL. Equivalent to
+    'docker kill <container> --signal <signal>'
+```
+
+> Note that signal groups consist of signals that fall under a specific category.
+
+`signal` currently works with Docker containers, AWS EC2 instances, and GCP compute instances (try it out!).
+ 
+**For plugin authors:** You can get `signal` to work with other entries by implementing the `signal` action (see [here]({{ 'docs/external-plugins#signal' | relative_url }}) for external plugins, [here](https://godoc.org/github.com/puppetlabs/wash/plugin#Signalable) for core plugins). Checkout the corresponding entry schema docs to see how you'd specify an entry's supported signals. Note that the `signal` action does _not_ have to wait for the given signal operation to finish. It is enough to return once the signal's been successfully received by the plugin's API. We also don't impose any restrictions on the sent signal's name, so you are free to name your signals however way you want.
+
+Note that we're open to extending the `signal` action's power depending on user demand. Some things we're considering are:
+
+* Extending `signal` to take-in signal-specific API options. This would enable something like `signal docker/containers/wash_tutorial_redis_1 stop` to accept all of `docker stop`'s options. In a more general sense, this lets `signal` take full advantage of the plugin's API by allowing the user to pass-in API-specific options.
+
+* Extending `signal` to return an (optional) result of the signaled operation. Combined with (1), this would enable `signal` to be the "kitchen-sink" Wash action, where plugin authors can specify all vendor-specific operations that are not captured by any of the other Wash actions as signals (similar to an arbitrary method invocation).
```

Comment:
This could be an interesting, formal definition of `signal` and would probably be a useful thing to support in the future.

Reply:
I'm not sure how useful that is as a standard pattern. Much of the value comes from having broadly useful, comprehensible commands. If signal becomes a grab-bag for anything I think it loses those characteristics.

Reply:
It can serve both purposes. This mostly came from trying to find a formal definition of signal. Started with "anything that has operational state", then the issue with that is maybe you may want to signal entries that don't necessarily have an operational state (or send a signal that may not change the operational state), so that means the definition has to be more general.